src/taipy/core/data/json.py

```diff
@@ -115,4 +116,13 @@ def _cast(self, data: Any):
 
     def _write(self, data: Any):
         with open(self._path, "w") as f:  # type: ignore
-            json.dump(data, f)
+            json.dump(data, f, indent=4, cls=EnhancedJSONEncoder)
+
+
+class EnhancedJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, (datetime, date)):
+            return o.isoformat()
+        if dataclasses.is_dataclass(o):
+            return dataclasses.asdict(o)
+        return super().default(o)
```

Comment:
If you create a custom encoder I believe you need to have a custom decoder as well. Other wise it cannot work properly.

Reply:
How about letting the user pass in the encoder and decoder?

Reply:
The custom decoder you mention about is the `exposed_type` property. However it cannot decode deeply nested objects.

Reply:
What is important is that when reading the data node, the user should retrieve the exact same json that he/she wrote.

In other words, the following should always pass whatever foo is: 
```
my_JSON_data_node.write(foo)
assert foo= my_JSON_data_node.read()

```
I believe we need to accept optional attributes in the data node config. Something like encoder/decoder, converter/exposed_type, or any other more meaningful names.